proof/invariant-abstract/AARCH64/ArchKHeap_AI.thy

```diff
@@ -817,6 +818,17 @@ lemma valid_arch_state_lift_aobj_at:
   apply simp
   done
 
+(* interface lemma *)
+lemma valid_arch_state_lift_aobj_at:
+  assumes caps: "\<And>P. \<lbrace>\<lambda>s. P (caps_of_state s)\<rbrace> f \<lbrace>\<lambda>_ s. P (caps_of_state s)\<rbrace>"
+  shows "f \<lbrace>valid_arch_state\<rbrace>"
+  unfolding valid_arch_state_def valid_asid_table_def valid_global_arch_objs_def pt_at_eq
+  apply (wp_pre, wps arch aobjs_of_lift_aobj_at)
+   apply (wpsimp wp: hoare_vcg_all_lift hoare_vcg_ball_lift vmid_inv_ap_lift
+                     aobjs_of_lift_aobj_at arch)
+  apply simp
+  done
```

Comment:
Fine if you can't be bothered, but this proof could just be applying `valid_arch_state_lift_aobj_at_no_caps`.

Reply:
tuned to this (also on RISCV64):
```
  assumes caps: "⋀P. f ⦃λs. P (caps_of_state s)⦄"
  shows "f ⦃valid_arch_state⦄"
  by (rule valid_arch_state_lift_aobj_at_no_caps)
```